tests/lib/cli-engine/config-array/ignore-pattern.js

```diff
@@ -52,74 +53,107 @@ describe("IgnorePattern", () => {
+                    ["foo/bar/dir/c.js", true],
+                    ["foo/bar/dir/c.ts", false],
+                    ["abc/a.js", false],
+                    ["abc/a.ts", false],
+                    ["abc/b.js", false],
+                    ["abc/b.ts", false],
+                    ["abc/c.js", true],
+                    ["abc/c.ts", true],
+                    ["abc/dir/a.js", false],
+                    ["abc/dir/a.ts", false],
+                    ["abc/dir/b.js", true],
+                    ["abc/dir/b.ts", false],
+                    ["abc/dir/c.js", true],
+                    ["abc/dir/c.ts", false]
+                ];
+
+                for (const [filename, expected] of patterns) {
+                    it(`should return ${expected} if '${filename}' was given.`, () => {
+                        assert.strictEqual(ignores(path.join(cwd, filename)), expected);
+                    });
+                }
+
+                it("should return false if '.dot.js' and false was given.", () => {
+                    assert.strictEqual(ignores(path.join(cwd, ".dot.js"), false), true);
                 });
-            }
 
-            it("should return false if '.dot.js' and false was given.", () => {
-                assert.strictEqual(ignores(path.join(cwd, ".dot.js"), false), true);
-            });
+                it("should return true if '.dot.js' and true were given.", () => {
+                    assert.strictEqual(ignores(path.join(cwd, ".dot.js"), true), false);
+                });
 
-            it("should return true if '.dot.js' and true were given.", () => {
-                assert.strictEqual(ignores(path.join(cwd, ".dot.js"), true), false);
-            });
+                it("should return false if '.dot/foo.js' and false was given.", () => {
+                    assert.strictEqual(ignores(path.join(cwd, ".dot/foo.js"), false), true);
+                });
 
+                it("should return true if '.dot/foo.js' and true were given.", () => {
+                    assert.strictEqual(ignores(path.join(cwd, ".dot/foo.js"), true), false);
+                });
+            }
 
-            it("should return false if '.dot/foo.js' and false was given.", () => {
-                assert.strictEqual(ignores(path.join(cwd, ".dot/foo.js"), false), true);
-            });
+            assertions(process.cwd());
+            assertions(path.parse(process.cwd()).root);
+        });
+    });
 
-            it("should return true if '.dot/foo.js' and true were given.", () => {
-                assert.strictEqual(ignores(path.join(cwd, ".dot/foo.js"), true), false);
-            });
+    describe("static createIgnore(ignorePatterns)", () => {
+        it("with common ancestor of drive root on windows should not throw", () => {
+            try {
+
+                // when not on windows return win32 values so local runs on linux produce the same issue as windows.
+                if (process.platform !== "win32") {
```

Comment:
Thanks for updating this. I think this is definitely an improvement. I still think it would be nice to ensure that execution is the same regardless of the platform the test is run on. Could we make this two tests - one for Windows-style paths and another for *nix-style paths - that don't conditionally stub these core Node APIs?

Reply:
line 141-145 - to be clear, with this the code execution of the module under test i.e ./tests/*/igore-pattern.js is the same when this test is run on linux and windows. The only portion of code that is not the same when running this test on linux/windows is the actual unit test i.e test/*/ignore-pattern.js. This is required to be this way as this is a platform specific bug and as such conditional platform specific stubs are required to ensure the tests exercises the same code path within the module under test irrespective of if you run the test on linux or windows.  If you don't stub these APIs to behave like windows then when running on linux then this test will not exercise the code path with the fix and will not catch future regressions.

line 132 - as for linux root path tests - / for *nix and C:\ for win32 they along with windows are still covered in line 132 with the original full set of ~40 assertions by `assertions(path.parse(process.cwd()).root);`.

I don't think there is more to do here. coverage has been increased to cover root path on linux and windows with line 132 by exercising the full original test suite against root path on linux./windows. and this new tests provides coverage and execution of the fix by exercising the same code path of the module under test when run on windows and linux with windows style paths.  if you believe there is more to do here please provide specific code example.